include/kodlab_mjbots_sdk/limb_base.h

```diff
@@ -0,0 +1,155 @@
+    };
+
+    class LimbBase
+    {
+    public:
+        /**
+         * @brief Construct a new Limb Base object with a name
+         *
+         * @param name          /// Sets the limb name
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::string name,
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        /**
+         * @brief Construct a new Limb Base object
+         *
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        virtual ~LimbBase(){};
+
+        /**
+         * @brief Update position and velocity of each joint within leg
+         *
+         * @param pos_list desired positions of each joint
+         * @param vel_list desired velocities of each joint
+         *
+         */
+        void Update(std::vector<float> pos_list, std::vector<float> vel_list);
+
+        /**
+         * @brief Calculates the forward kinematics of the leg
+         *
+         * @return Eigen::VectorXf
+         */
+        virtual void ForwardKinematics() = 0;
+
+        /**
+         * @brief Calculates the Jacobian of the leg
+         *
+         * @return Eigen::MatrixXf
+         */
+        virtual void Jacobian() = 0;
+
+        /**
+         * @brief Calculates the Inverse kinematics of a leg given a desired position
+         * measured from the origin of the leg (????)
+         *
+         * @param x_des desired x position
+         * @param y_des desired y position
+         * @param z_des desired z position
+         *
+         * @return std::vector<float> positions
+         */
+
+        virtual void InverseKinematics(std::vector<float> EE_pos) = 0;
+
+        /**
+         * @brief Get the Positions of each joint in the leg
+         *
+         * @return std::vector<float>
+         */
+        std::vector<float> get_positions();
+
+        /**
+         * @brief Get the velocities of each joint in the leg
+         *
+         * @return std::vector<float>
+         */
+        std::vector<float> get_velocities();
+
+        /**
+         * @brief Get the Torques of each joint in the leg
+         *
+         * @return std::vector<float>
+         */
+        std::vector<float> get_torques();
+
+        /**
+         * @brief Set the positions of each joint in the leg
+         *
+         * @param positions
+         */
+
+        void set_positions(std::vector<float> positions);
+
+        /**
+         * @brief Set the velocities of each joint in the leg
+         *
+         * @param velocities
+         */
+
+        void set_velocities(std::vector<float> velocities);
```

Comment:
Pass by constant reference per [ISO C++ reference](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information):
```suggestion
        void set_velocities(const std::vector<float> &velocities);
```

Reply:
fixed for larger formatting commit